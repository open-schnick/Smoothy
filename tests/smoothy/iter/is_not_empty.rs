use crate::failing_assertion;
use smoothy::{assert_that, AssertionConnector, EqualityAssertion, IteratorAssertion};

#[test]
fn succeeds() {
    let vec: Vec<String> = vec![String::from("Hello World!")];
    let _connector: AssertionConnector<Vec<String>> = assert_that(vec).is_not_empty();
}

#[test]
fn connects_with_other_iter_assertions() {
    let vec = ["A", "B", "C"];

    assert_that(vec).is_not_empty().and().second().equals("B");
}

#[test]
fn fails() {
    failing_assertion!({
        let vec: Vec<String> = vec![];
        assert_that(vec).is_not_empty();
    });
}
