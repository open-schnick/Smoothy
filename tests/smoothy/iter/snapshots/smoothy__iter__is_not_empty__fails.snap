---
source: tests/smoothy/iter/is_not_empty.rs
expression: assertion_failed_output
---
[31mAssertion failed![0m

Expected
  [31m[][0m
to contain at least one item
